### PR TITLE
Install miniconda in VSO image and build apps targeting conda-jupyter

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: images/build/Dockerfiles/vso.Dockerfile [hunk 1]
```diff
@@ -6,6 +6,30 @@ ARG GIT_COMMIT=unspecified
...
+RUN install -o root -g root -m 644 conda.gpg /usr/share/keyrings/conda-archive-keyring.gpg
+RUN gpg --keyring /usr/share/keyrings/conda-archive-keyring.gpg --no-default-keyring --fingerprint 34161F5BF5EB1D4BFBBB8F0A8AEB4F8B29D82806
+RUN echo "deb [arch=amd64 signed-by=/usr/share/keyrings/conda-archive-keyring.gpg] https://repo.anaconda.com/pkgs/misc/debrepo/conda stable main" > /etc/apt/sources.list.d/conda.list
+RUN . /tmp/__condaConstants.sh \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+        conda=${CONDA_VERSION} \
+    && rm -rf /var/lib/apt/lists/*
+ENV CONDA_SCRIPT="/opt/conda/etc/profile.d/conda.sh"
+RUN . $CONDA_SCRIPT && \
```
Comment (arroyc) [inline images/build/Dockerfiles/vso.Dockerfile:24]: @kichalla .. can we please consolidate all these run statements into one or two run statements by using bash script or something? so from line 10 to line 22

Comment (kichalla) [reply images/build/Dockerfiles/vso.Dockerfile:24]: Sure, will do.

File: src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs [hunk 2]
```diff
@@ -264,13 +264,24 @@ public bool IsEnabledForMultiPlatformBuild(RepositoryContext ctx)
...
+        /// <inheritdoc/>
         public string ResolveVersion(string versionToResolve)
         {
             var resolvedVersion = GetVersionUsingHierarchicalRules(versionToResolve);
             resolvedVersion = GetMaxSatisfyingVersionAndVerify(resolvedVersion);
             return resolvedVersion;
         }
 
+        /// <inheritdoc/>
+        public IDictionary<string, string> GetToolsToBeSetInPath(
```
Comment (qianz2) [inline src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs:276]: Could we make the `detectorResult `global in this case and access from `DefaultPlatformsInformationProvider`

File: src/BuildScriptGenerator/DefaultPlatformsInformationProvider.cs [hunk 3]
```diff
@@ -52,26 +51,34 @@ public IEnumerable<PlatformDetectorResult> DetectPlatforms(RepositoryContext con
                 }
 
                 var detectionResult = platform.Detect(context);
+
                 if (detectionResult != null)
                 {
-                    detectionResults.Add(detectionResult);
+                    var toolsInPath = platform.GetToolsToBeSetInPath(context, detectionResult);
+
+                    platformInfos.Add(new PlatformInfo
```
Comment (qianz2) [inline src/BuildScriptGenerator/DefaultPlatformsInformationProvider.cs:59]: It looks like the platformInfo has a detectionResult and a dictionary of its content. wondering if we can simplify this if we only want to have them printed out.

Comment (kichalla) [reply src/BuildScriptGenerator/DefaultPlatformsInformationProvider.cs:59]: Let us discuss this in today's meeting.

**Pull request merged**